airbyte-integrations/connectors/source-stock-ticker-api-cdk/source_stock_ticker_api_cdk/source.py

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from datetime import datetime, timedelta, timezone
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import NoAuth
+
+
+class StockPrices(HttpStream):
+    url_base = "https://api.polygon.io/v2/aggs/ticker/"
+    cursor_field = "date"
+    primary_key = "date"
+
+    def __init__(self, config: Mapping[str, Any], start_date: datetime, **kwargs):
+        super().__init__()
+        self.api_key = config["api_key"]
+        self.stock_ticker = config["stock_ticker"]
+        self.start_date = start_date
+        self._cursor_value = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # The API does not offer pagination, so we return None to indicate there are no more pages in the response
+        return None
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"{self.stock_ticker}/range/1/day/{stream_slice['start_date']}/{stream_slice['end_date']}"
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        # The api requires that we include apikey as a header so we do that in this method
+        return {"Authorization": f"Bearer {self.api_key}"}
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        # The api requires that we include the base currency as a query param so we do that in this method
+        return {"sort": "asc", "limit": 120}
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        # The response is a simple JSON whose schema matches our stream's schema exactly,
+        # so we just return a list containing the response
+        response_dict = response.json()
+        data = []
+        if response_dict["resultsCount"]:
+            for day_result in response_dict["results"]:
```

Comment:
If the 'resultsCount' dictionary does not contain a 'results' key, we encounter an error without a human-readable error message.

Reply:
Fixed